sway-ir/src/irtype.rs

```diff
@@ -197,12 +205,20 @@ impl Type {
             // Unions are special.  We say unions are equivalent to any of their variant types.
             (_, TypeContent::Union(_)) => other.eq(context, self),
             (TypeContent::Union(l), _) => l.iter().any(|field_ty| other.eq(context, field_ty)),
+            // Never type can coerce into any other type.
+            (TypeContent::Never, _) => true,
```

Comment:
I am not sure about this. `Never` can coerce into any other type but not other way around. Also, I am not sure if `eq` on the IR side is about coercion at all or if it checks strict equality of types. Before analyzing the usage, @vaivaswatha can you please take a look and give your view.